framework/behaviors/CacheableBehavior.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+namespace yii\behaviors;
+
+use yii\base\Behavior;
+use yii\base\Widget;
+use yii\base\WidgetEvent;
+use yii\caching\Cache;
+use yii\caching\Dependency;
+use yii\di\Instance;
+use yii\web\View;
+
+/**
+ * Cacheable widget behavior.
+ *
+ * @property Widget $owner
+ *
+ * @group behaviors
+ */
+class CacheableBehavior extends Behavior
+{
+    /**
+     * The cache object or the application component ID of the cache object
+     * or a configuration array for creating the object.
+     *
+     * @var Cache|string|array
+     */
+    public $cache = 'cache';
+
+    /**
+     * Cache duration in seconds.
+     *
+     * @var int
+     */
+    public $cacheDuration = 60;
```

Comment:
If this is a cacheable behavior, we do expect that everything inside of this class will be related to cache, so I think it's better to remove "cache" prefixes from properties, for example instead of `$cacheKey` use just `$key` and so on.